app/src/main/java/com/hedvig/app/feature/keygear/ui/tab/KeyGearViewModel.kt

```diff
@@ -14,16 +14,40 @@ import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.launch
 
 abstract class KeyGearViewModel : ViewModel() {
-    abstract val data: LiveData<KeyGearItemsQuery.Data>
+    abstract val data: LiveData<Result<KeyGearItemsQuery.Data>>
 
     abstract fun sendAutoAddedItems()
+    abstract fun load()
 }
 
 class KeyGearViewModelImpl(
     private val repository: KeyGearItemsRepository,
     private val deviceInformationService: DeviceInformationService
 ) : KeyGearViewModel() {
-    override val data = MutableLiveData<KeyGearItemsQuery.Data>()
+    override val data = MutableLiveData<Result<KeyGearItemsQuery.Data>>()
+
+    init {
+        load()
+    }
+
+    override fun load() {
+        viewModelScope.launch {
+            repository
+                .keyGearItems()
+                .onEach { response ->
```

Comment:
It would be cool to throw an error in the repository if response.errors != null, instead of handling errors in two places here